backend/src/domains/user/user.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { UserEntity } from './user.entity';
+import { CreateUserDto } from "./dto/create-user.dto"
+import { UpdateUserDto } from "./dto/update-user.dto"
+import * as bcrypt from 'bcryptjs';
+
+@Injectable() 
+export class UserService {
+  constructor(@InjectRepository(UserEntity) private readonly UserRepository: Repository<UserEntity>) { }
+  private saltRounds = 10;
+
+  async createUser (data: CreateUserDto): Promise<UserEntity> {
+    let user = new UserEntity();
+    user.login = data.login;
+    user.password = await this.getHash(data.password);
+//    user.password = data.password;
+    user.createdAt = new Date();
```

Comment:
Each time we have to make this operation by hand? It will be very sad. We need to find a way to automate it.

Reply:
Which operation do you mean? Do you mean "user.createdAt = new Date()"? This date should be inserted automatically (@CreateDateColumn) if the entity is inherited from the base entity. But this entity (UserEntity) is not inherited from the Base Entity. Therefore "new Date()" called by hand.

Reply:
Why the User entity doesn't inherited from the base etntity?